src/normalize/mod.rs

```diff
@@ -57,3 +61,50 @@ pub use user::UserOutput;
 
 pub(crate) use user::normalize_sorted_user_addrs_with_entries;
 pub(crate) use user::Handler;
+
+
+/// The reason why normalization failed.
+///
+/// The reason is generally only meant as a hint. Reasons reported may change
+/// over time and, hence, should not be relied upon for the correctness of the
+/// application.
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+#[non_exhaustive]
+pub enum Reason {
+    /// The absolute address was not found in the corresponding process' virtual
+    /// memory map.
+    Unmapped,
+    /// The `/proc/<pid>/maps` entry corresponding to the address does not have
+    /// a component (file system path, object, ...) associated with it.
+    MissingComponent,
+    /// The address belonged to an entity that is currently unsupported.
+    Unsupported,
```

Comment:
Can we report the entry as a String? This prevents the user to go and dig into the proc maps file to figure out what was wrong.

```
Unsupported(String)
```

Reply:
We do not want to allocate this part and keep this type light weight. We can include more details in traces.